source/world_builder/features/subducting_plate_models/temperature/mass_conserving.cc

```diff
@@ -260,26 +273,46 @@ namespace WorldBuilder
+                                                           ridge_spreading_velocities_at_each_ridge_point,
                                                            world->parameters.coordinate_system,
-                                                           trench_point_natural);
+                                                           trench_point_natural,
+                                                           subducting_velocities_at_each_ridge_point);
 
               const double km2m = 1.0e3; // 1000 m/km
               const double cm2m = 100; // 100 cm/m
               const double my = 1.0e6;  // 1e6 y/my
-              const double seconds_in_year = 60.0 * 60.0 * 24.0 * 365.25;  // sec/y
-              const double plate_velocity = ridge_parameters.first * seconds_in_year; // m/yr
-
-              const double age_at_trench = ridge_parameters.second / plate_velocity; // m/(m/y) = yr
-              const double plate_age_sec = age_at_trench * seconds_in_year; // y --> seconds
 
               /* information about nearest point on the slab segment */
               const double distance_along_plane = distance_from_planes.distance_along_plane;
               const double depth_to_reference_surface = distance_from_planes.depth_reference_surface;
               const double total_segment_length = additional_parameters.total_local_segment_length;
               const double average_angle = distance_from_planes.average_angle;
 
+              const double seconds_in_year = 60.0 * 60.0 * 24.0 * 365.25;  // sec/y
+              const double spreading_velocity = ridge_parameters[0] * seconds_in_year; // m/yr
+              const double subducting_velocity = ridge_parameters[2] * seconds_in_year; // m/yr
+              const double ridge_velocity = spreading_velocity - subducting_velocity; // m/yr
+              const int sign_v = (ridge_velocity > 0) ? 1 : ((ridge_velocity < 0) ? -1 : 0);
+
+              const double time_of_ridge_migration = subducting_velocities.first[0]; // yr
+              const double ridge_drift_distance = abs(ridge_velocity) * time_of_ridge_migration; // m
+
+              double effective_age_shift = 0;
+              double trench_age_shift = 0;
+              if (subducting_velocity != 0 && ridge_velocity != 0)
+                {
+                  if (distance_along_plane < ridge_drift_distance)
+                    {
+                      effective_age_shift = (1 - (distance_along_plane / ridge_drift_distance)) * sign_v * \
+                                            distance_along_plane / subducting_velocity; // yr
+                      trench_age_shift = (1 - (distance_along_plane / ridge_drift_distance)) * sign_v * \
+                                          distance_along_plane / spreading_velocity; // yr
+                    }
+                }
+
+              const double age_at_trench = ridge_parameters[1] / spreading_velocity + trench_age_shift; // m/(m/y) = yr
```

Comment:
My main comment is: is there a way to walk around the effective_age_shift ? this variable is not very intuitive and it takes me several times to learn its meaning.
At the same time, I am confused about why we need the time_of_ridge_migartion in the first place.
Maybe it's possible to use the subducting time and:
maybe we can define a parameter of ridge distance as:
`ridge_distance = ridge_velocity * time_of_subduction`
then the age_at_trench could be modified as
"age_at_trench = ridge_parameters[1] / spreading_velocity  + ridge_distance / ridge_velocity

Reply:
The names of the variables could definitely use some improvement. But what I intend `effective_age_shift` to be is a value that increases/decreases the plate age after it has been subducted, white `trench_age_shift` is a value that increases/decreases what the age of the plate is at the trench.

The idea for why I made `time_of_ridge_migration` a variable is so that the user can define how long the ridge has been moving, but I'm certainly open to ideas! It's not clear to me what you mean by `time_of_subduction`, though. The time that subduction started?